packages/devtools_app/lib/src/screens/debugger/program_explorer_controller.dart

```diff
@@ -105,18 +109,34 @@ class ProgramExplorerController extends DisposableController
     ScriptRef script,
     List<VMServiceObjectNode> nodes,
   ) {
+    final searchCondition = (node) => node.script?.uri == script.uri;
     for (final node in nodes) {
-      final result = node.firstChildWithCondition(
-        (node) => node.script?.uri == script.uri,
-      );
+      final result = node.firstChildWithCondition(searchCondition);
       if (result != null) {
         selectNode(result);
         result.expandAscending();
+        _selectedNodeIndex.value = _calculateNodeIndex(searchCondition);
         return;
       }
     }
   }
 
+  int _calculateNodeIndex(
+      bool matchingNodeCondition(VMServiceObjectNode node)) {
+    var index = 0;
+    for (final node in _rootObjectNodes.value) {
+      final matchingNode = depthFirstTraversal(
+        node,
+        returnCondition: matchingNodeCondition,
+        exploreChildrenCondition: (node) => node.isExpanded,
```

Comment:
what if the file uri we are looking for is in a collapsed part of the tree?

Reply:
We are currently only calling this method directly after expand-ascending up from the selected file, but this is a good point. I've added a parameter to this method to specify whether or not to search in collapsed parts of the tree, that way if it gets re-used we aren't accidentally skipping nodes in the search.